spotifycli/test/test_spotifycli.py

```diff
@@ -0,0 +1,163 @@
+from io import StringIO
+from pathlib import Path
+from subprocess import PIPE
+from unittest import skip
+from unittest.mock import patch
+
+import dbus
+import dbusmock
+
+from spotifycli.spotifycli import main
+from spotifycli.version import __version__
+
+
+class TestSpotifyCLI(dbusmock.DBusTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.start_session_bus()
+        cls.dbus_con = cls.get_dbus(system_bus=False)
+
+    def setUp(self):
+        self.dbus_spotify_server_mock = self.spawn_server(
+            "org.mpris.MediaPlayer2.spotify",
+            "/org/mpris/MediaPlayer2",
+            "org.freedesktop.DBus.Properties",
+            system_bus=False,
+            stdout=PIPE
+        )
+        self.dbus_spotify_interface_mock = dbus.Interface(
+            self.dbus_con.get_object(
+                'org.mpris.MediaPlayer2.spotify', '/org/mpris/MediaPlayer2'
+            ),
+            dbusmock.MOCK_IFACE
+        )
+        metadata_file = Path(__file__).resolve().parent / 'dbus_metadata_response.py'
+        self.dbus_spotify_interface_mock.AddMethod(
+            '', 'Get', 'ss', 'v',
+            metadata_file.read_text(),
+        )
+        # TODO: Mock up the controls bus/interface, since I think it is slightly different
+        # self.dbus_spotify_interface_mock.AddMethod(
+        #     '', 'Play', '', '', ''
+        # )
+
+    def tearDown(self):
+        self.dbus_spotify_server_mock.stdout.close()
+        self.dbus_spotify_server_mock.terminate()
+        self.dbus_spotify_server_mock.wait()
+
+    def test_cli_usage(self):
+        with patch('sys.argv', ["spotifycli", "--help"]), patch("sys.exit") as mock_exit, patch('sys.stdout', new_callable=StringIO) as buffer:
+            main()
+            self.assertIn("usage", buffer.getvalue())
+            mock_exit.assert_called_with(0)
```

Comment:
OK, here's a whole list of individual unit tests that test each of the different command line options.  I'll describe it here once in general:

- Each test starts with a `with` block, where we utilize some patches to adjust behavior for the unit test:
  - `patch('sys.argv', ["spotifycli", "--help"])` will override the values of sys.argv for the whole test so that the argparse library sees it the way we want.
  - `patch("sys.exit") as mock_exit` will capture `sys.exit` calls so we can try to make sure the process succeeded as it is about to exit back to the shell
  - `patch('sys.stdout', new_callable=StringIO) as buffer` creates a new string buffer that overrides `sys.stdout`, where we can check to see what was printed to the command line window.
- We then call `main`, which will operate based on the patched argv
- We then do assertions on the call result, or whether certain strings are found, etc.